src/ddmd/mtype.d

```diff
@@ -3075,6 +3075,13 @@ extern (C++) abstract class Type : RootObject
     {
         v.visit(this);
     }
+
+    final TypeFunction toTypeFunction()
+    {
+        if (ty != Tfunction)
+            assert(0);
```

Comment:
@WalterBright It'll be useful to someone debugging a problem in the compiler.
If you insist on assert(0) using a proper error messages is even more important such that the user does atleast have something to put into bugzilla when reporting the crash.